server/workflow_runner/validators.py

```diff
@@ -112,13 +136,18 @@ def _validate_field(row_num: int, row: dict, field: FieldSchema, params: dict[st
         case BasicFieldDataTypeSchema(data_type="number"):
             try:
                 float(value)
-            except ValueError:
-                validations.append(
-                    ValidationFailure(
-                        row_number=row_num,
-                        message=f"Value '{value}' for field '{field.name}' is not a valid number"
+            # TypeError is raised if value is None
+            except (TypeError, ValueError) as e:
```

Comment:
Idk if this has to do with my python interpreter (i'm using the venv), because I don't know how the original unit tests didn't see this. 

But basically the float cast throws a TypeError if the input is None (which happens when there are missing cells), which isn't caught by the original except statement.